os/pm/pm_activity.c

```diff
@@ -130,6 +132,27 @@ void pm_activity(int domain, int priority)
 
 		pdom->accum = (int16_t)accum;
 
+	#ifdef CONFIG_PM_METRICS
+		DEBUGASSERT(callbacks);
+
+		FAR sq_entry_t *node = NULL;
+		int flag = ERROR;
+		for (node = sq_peek(&(pdom->each_accum)); node; node = node->flink) {
+			if (!strncmp(((struct pm_accumchange_s*)node)->name, callbacks->name, sizeof(callbacks->name))) {
```

Comment:
How about changing this like below?
struct pm_accumchange_s* node;
node = (struct pm_accumchange_s*)sq_peek(&(pdom->each_accum);
strncmp(node->name, ...